Minor edits to documentation (#519)
Kitware/TeleSculptor: doc/glossary.rst

  the term "frame" is used to indicate
  instances where the pose may not yet be known.

Camera Pose:
  In TeleSculptor, a camera pose

>   (often shortened to "camera")

Reviewer: ...or at least a) fix instances of "pose" that I inserted due to trying to use that as per the previous version, and b) add some text specifically calling out that we are using the term in such manner. (Also, we should try to use "physical camera" in some places, e.g. "parameters of a physical camera". I suppose that suggests "virtual camera" as an alternative term?)

>   (often shortened to "camera")

Author: Yes, the problem is that camera pose is just the position and orientation.  When we say "camera" we usually mean a model of both pose and intrinsic parameters.  We usually call them just "cameras" but I suppose "camera models" is more accurate.

>   (often shortened to "camera")

Reviewer: "Camera models" works. We should also check what other uses of "pose" should be "model". My main objective was to not use "camera", which is too easily confused with "*physical* camera", i.e. something of which you only have *one* if your input is video or a collection of images taken with the same device and lens.

  which describes the intrinsic and extrinsic parameters of a camera

>   at a given frame.

Author: no objection , but not sure what your wrapping rule is here.  I was just wrapping at 80 characters or end of sentence.

>   at a given frame.

Reviewer: Right; not that. Since the line breaks don't matter, it's better to wrap in a way that minimizes the likelihood of needing to rewrap if the wording is changed. I already partly mentioned this elsewhere, but to be more complete:
- Keep lines under 80¹ characters (obviously).
- *Always* wrap after a sentence.
- Everywhere else, try to wrap where a pause "makes sense".
- Keeping line lengths similar is usually preferable, but not at the expense of the preceding rules.

For the second-to-last, after commas is usually a good place. I will sometimes break after each item in a list (should help with diff churn if an item needs to be added or removed). Otherwise, thinking about where a pause would be least awkward if the text was spoken aloud seems to be a good "rule of thumb". (I'll sometimes "fudge" this a bit for the sake of the last rule, but breaking somewhere that would clearly be an awkward verbal pause is to be avoided.)

In a sense, it's not that different from how we would break if it was code. (Break after statements, prefer to keep related things together, strive for similar line lengths but not at the expense of logical flow.)

(¹ As always, that's also pedantically "79 characters or less, unless you really, really have to have 80".)

  Intrinsic parameters include focal length and lens distortion.
  Extrinsic parameters are the camera pose |--| orientation and position.
  A frame may have at most one camera pose,
  which is unique to that frame.
  While extrinsic parameters are never shared across frames,
  intrinsic parameters may or may not be share depending on the algorithm
  that produced them.

  In the GUI, cameras are represented as rectangular pyramids.

>   The peak of the pyramid is the camera center of projection.

Author: I think all three terms are equally valid

  The rectangle base is proportional to the image aspect ratio,
  and a triangle attached to the base indicates
  which direction is "up" in the image.
  The pyramid visualizes the field of view of the camera.

> 

Reviewer: Okay (when I wrote this, I hadn't realized there *is* an option to not force shared intrinsics), but please squash the consecutive blank lines.

Feature:
  A feature is a location that corresponds to a salient point in an image.

>   Features have enough visual texture that they can be reliably localized
>   in images across time.

Reviewer: Weak suggestion; I'm not absolutely in love with the line breaks, but not sure if this is an improvement.
```suggestion
  Features have enough visual texture
  that they can be reliably localized
  in images across time.
```

>   Features are also known as interest points or corner points.

Reviewer: Are these terms actually used in photogrammetry? I would find "corner points" on its own is ambiguous. If you're just trying to preserve what I had written previously, I think it would be better to just drop this sentence in favor of what you wrote about "texture".

Author: They are actually called "interest points" and "corner points" in other parts of the literature

Track:

>   A track, or feature track, is a collection of correlated features over time;

Reviewer: ```suggestion
  A (feature) track is a collection of correlated features;
```
"Time" is a fairly artificial construct in TeleSculptor that is largely or entirely irrelevant to reconstruction. If you really feel it necessary to say something, I would say something more like "across multiple frames".

  that is, detected feature points
  estimated to correspond to the same landmark.

Landmark:
  A landmark is an estimated 3D world point that,
  when projected into the images,
  gave rise to an observed feature track.

Residual:
  A residual, in general, is the difference between an observed value and an
  estimated value\ [#er]_. In TeleSculptor, the observed value is typically a
  detected feature point, and the estimated value is the projection of its

>   corresponding landmark into the image.

Reviewer: Ack, this whole paragraph didn't get rewrapped.

Should be something like:
```
  A residual, in general, is the difference
  between an observed value and an estimated value\ [#er]_.
  In TeleSculptor, the observed value
  is typically a detected feature point,
  and the estimated value is the projection
  of its corresponding landmark into the image.
```
(Wording is fine, though! Just the line wraps, and that's partly my fault.)

Author: I'm really not that worried about the wrapping as long as the lines are not too long.

Reviewer: As noted, it makes for better diffs when the text gets revised...

Ground Control Point (GCP):

>   A ground control point, or GCP, is a 3D point that is
>   manually placed in the scene by a user.
>   GCPs are often attached to an identifiable location
>   from a reference image or map.
>   The user can assign the true geodetic location
>   (latitude, longitude, elevation) to a GCP,
>   and that GCP can then serve as a constraint when geo-registering the model.

Reviewer: ```suggestion
  A |gcp| is a 3D point
  that is manually placed in the scene by a user.
  Ground control points are often attached 
  to an identifiable location from a reference image or map.
  The user can assign the true geodetic location
  (latitude, longitude, elevation) to a |gcp|,
  and that |gcp| can then serve as a constraint 
  when geo-registering the model.
```
(In general, I'd prefer to avoid abbreviating GCP in the documentation. It saves very little typing, except at the beginning of sentences. Likewise for CRPs.)

Camera Registration Point (CRP):

>   A camera registration point, or CRP, is a 2D point that is
>   manually placed in one or more images by a user.
>   CRPs are associated with a GCP point and can tie that GCP
>   to multiple images, even when no camera model has been estimated.
>   While a GCP is the manually placed version of a landmark,
>   a CRP is the manually placed version of a feature track.
>   A collection of CRPs can be used to manually estimate a camera model
>   when automated feature matching is not possible.

Reviewer: ```suggestion
  A |crp| is a 2D point 
  that is manually placed by the user
  in one or more frame images.
  Camera registration points are associated with a |gcp|
  and can tie that |gcp| to multiple images,
  even when no camera model has been estimated.
  While a |gcp| is the manually placed version of a landmark,
  a |crp| is the manually placed version of a feature track.
  A collection of |crp|\ s can be used 
  to manually estimate a camera model
  when automated feature matching is not possible.
```
I'm somewhat inclined to write "where[as] a GCP..." instead of "while", but not sure; thoughts? (Also, ugh, please don't write "GCP point" 🙂... "ATM machine", "Mount Midoriyama", etc. 😒)

Author: either is fine with me.
